File: 2026-08/src/day_29/prms_181894/Solution.java
package day_29.prms_181894;

import java.util.*;

class Solution {
    public int[] solution(int[] arr) {
        int n = arr.length;
        int s = 0, e = n - 1;

        while (s < n && arr[s] != 2) {
            s++;
        }
        if (s == n) return new int[]{-1};

        while (arr[e] != 2) {
            e--;
        }

        return Arrays.copyOfRange(arr, s, e + 1);
    }
}

File: 2026-08/src/day_29/prms_181894/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> solution(vector<int> arr) {
    int n = arr.size();
    int s = 0, e = n - 1;

    while (s < n && arr[s] != 2) s++;
    if (s == n) return {-1};

    while (arr[e] != 2) e--;

    return vector<int>(arr.begin() + s, arr.begin() + e + 1);
}
